x/ethbridge/types/msgs.go

```diff
@@ -207,29 +242,45 @@ func ComputeProphecyID(cosmosSender string, sequence uint64, ethereumReceiver st
 		{
 			Type: addressTy,
 		},
-		// {
-		// 	Type: uint256Ty,
-		// },
-		// {
-		// 	Type: boolTy,
-		// },
-		// {
-		// 	Type: uint128Ty,
-		// },
 		{
 			Type: uint256Ty,
 		},
+		{
+			Type: stringTy,
+		},
+		{
+			Type: stringTy,
+		},
+		{
+			Type: uint8Ty,
+		},
+		{
+			Type: int32Ty,
+		},
+		{
+			Type: boolTy,
+		},
+		{
+			Type: uint256Ty,
```

Comment:
the claim data struct in solidity labels nonce as a uint128 why did you change this to uint256?

Reply:
there is no uint128 type in go, so I use the big.Int as big.NewInt(int64(globalNonce)). In abi argument, we need a uint256 to pack it.